src/features/campaigns/hooks/useCreateCampaignActivity.ts

```diff
@@ -21,6 +23,18 @@ interface UseCreateCampaignActivityReturn {
   createSurvey: (surveyBody: ZetkinSurveyPostBody) => IFuture<ZetkinSurvey>;
 }
 
+async function fixNewCallAssignmentFilterConfig(
+  apiClient: IApiClient,
+  callAssignment: ZetkinCallAssignment
+): Promise<ZetkinCallAssignment> {
+  await apiClient.rpc(fixNewCallAssignmentFilterConfigRpc, {
+    callAssignmentId: callAssignment.id,
+    orgId: callAssignment.organization.id,
+    queryId: callAssignment.target.id,
```

Comment:
This should be using `callAssignment.goal.id` (instead of `target.id`) wherever it's happening (which IMO should be inside an RPC that first creates the assignmend and then updates it's goal query).